### PR TITLE
This commit contains improved voice-quality-feature extraction and bu…

### PR DESCRIPTION
**Pull request by fahim-zubayer**
…g-fixes based on

stress testing through voxforge dataset.

### CODE REVIEW

File: src/feat/feature-voice-quality-extractor.cc [hunk 1]
```diff
@@ -243,15 +242,18 @@ namespace kaldi
...
         for(int32 r = 0; r < numFrames; ++r )
         {
             const BaseFloat povFeat = povLocalCalculator(r, pitchFeatures);
-            logHNRLocalCalculator(r, pitchFeatures);
+            const BaseFloat logHNR = logHNRLocalCalculator(r, pitchFeatures);
+            mFeatures.logHNRLocal.push_back(logHNR);
+            mFeatures.povLocal.push_back(povFeat);
 
-            bool voiced_continuity = (r != 0) && (pitchFeatures(r - 1, 0) > mVqOpts.min_CC) &&
+            bool voiced_continuity = (r != 0) &&
```
Comment (fahim-zubayer) [inline src/feat/feature-voice-quality-extractor.cc]: Relaxation of the constraint to allow for more frames.

File: src/featbin/compute-kaldi-voice-quality-feature.cc [hunk 2]
```diff
@@ -110,7 +110,7 @@ int main(int argc, char *argv[])
         // VoiceQualityFeaturesExtractor Class computes all the features
         VQFeature features = extractor->computeFeatures(pitchFeatures, waveform);
 
-        int32 numFrames = pitchFeatures.NumRows();
+        int32 numFrames = features.povLocal.size();
```
Comment (shkyang) [inline src/featbin/compute-kaldi-voice-quality-feature.cc]: ok, so in Rubik, this is where you'd do the check for # of frames and stop processing

File: src/feat/voice-quality-structure.h [hunk 3]
```diff
@@ -20,5 +20,10 @@ namespace kaldi
             shimmerLocal.clear();
             logHNRLocal.clear();
         }
+        
+        VQFeature()
+        {
+            clearVectors();
```
Comment (shkyang) [inline src/feat/voice-quality-structure.h]: this seems unnecessary, stl vectors are initialized to be 0 length.

File: src/feat/voice-quality-structure.h [hunk 4]
```diff
@@ -11,7 +11,7 @@ namespace kaldi
         std::vector<BaseFloat> shimmerLocal; 
         std::vector<BaseFloat> logHNRLocal;
         
-        VQFeature()
+        void clearVectors()
```
Comment (shkyang) [inline src/feat/voice-quality-structure.h]: I'd get rid of this function

File: src/feat/feature-voice-quality-statistics.cc [hunk 5]
```diff
@@ -26,7 +26,7 @@ void VoiceQualityStatisticsComputer::calcVariation(const vector<BaseFloat> &para
     // we use 1 here because variance divides by params.size() -1
     if (params.size() <= 1)
     {
-        throw std::runtime_error("Error in calcVariation, input vector is zero length.");
+        throw std::runtime_error("Error in calcVariation, input vector is zero length. Possible cause : audio file did not produce sufficent frames under current configuration.");
```
Comment (shkyang) [inline src/feat/feature-voice-quality-statistics.cc]: get rid of input vector is zero length, get rid of Possible cause

Comment (shkyang) [inline src/feat/feature-voice-quality-statistics.cc]: Leave the following: 

Error in calcVariation: audio file did not produce sufficent frames.

File: src/feat/feature-voice-quality-extractor.h [hunk 6]
```diff
@@ -16,7 +16,7 @@ namespace kaldi
 	{
     protected:
         BaseFloat povLocalCalculator(int frameNo, const Matrix<BaseFloat> &pitchFeatures);
-        void logHNRLocalCalculator(int frameNo, const Matrix<BaseFloat> &pitchFeatures);
+        BaseFloat logHNRLocalCalculator(int frameNo, const Matrix<BaseFloat> &pitchFeatures);
```
Comment (shkyang) [inline src/feat/feature-voice-quality-extractor.h]: I think this is a good change, I like that the function does 1 thing, and doesn't push stuff to mPovLocal...

**Pull request closed without merging**